src/main/kotlin/no/nav/familie/ef/sak/service/GrunnlagsdataService.kt

```diff
@@ -65,13 +66,16 @@ class GrunnlagsdataService(private val registergrunnlagRepository: Registergrunn
             ?: it.søknadsgrunnlag.fødselTermindato
         }
 
+        val sagtOppEllerRedusertStilling = SagtOppEllerRedusertStillingMapper.tilDto(situasjon = søknad.situasjon)
```

Comment:
Burde alt som har med "dto" / infrastruktur-ting å gjøre holdes unna service klasser i ef-sak ? Eller er det greit at infrastruktur flyter litt sammen med business logikk her ? Er det tenkt at serviceklasser bare skal bestå av business logikk ?

Reply:
(lurer på litt det samme selv med en serviceklasse jeg driver med nå)

Reply:
Godt spørsmål, og jeg vet ikke hva som er best. Men i mitt hode er service-klassers rolle å sy sammen sammen business logikk og tilby funksjonalitet på en måte som kan brukes _direkte_ av APIet/interfacet. Og da kreves vel litt infrastruktur typ dto/mapping.

Reply:
Jeg ser i hvert fall ikke helt hensikten med å ha en puritansk tilnærming til business logikk og servicer. Et eller annet sted må vel infrastruktur blandes med domene/business logikk?

Reply:
For meg så har controller og repository klare avgrensninger på arbeidsansvar:
- Controller: tilgangskontroll, utfør en handling og returner svar basert på handlingen
- Repository: hent eller endre data i databasen

Problemet er litt det i mellom her - *service* er en sekkebetegnelse på det som sammenstimler data og konverterer opp og ned, stedet hvor logikken ofte blir implementert og validering av at operasjoner er lovlige. 
Med interface-baserte repository blir det til og med litt "repo"-kode inni disse servicene ettersom det ikke er mulig å lage implementasjon inni interface-repoene våre... 

Trenger vi et "nytt nivå"/lag ? Isåfall hva?